### PR TITLE
♻️ Strict Typescript

### PR DESCRIPTION
**Pull request by manuel-rw**

### CODE REVIEW

**Comment by manuel-rw** [comment]
FYI: The SonarCloud can be ignored. I am migrating to https://deepsource.io/ anyway.

File: src/clients/discord/discord.config.service.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import { GatewayIntentBits } from 'discord.js';
 export class DiscordConfigService implements DiscordOptionsFactory {
   createDiscordOptions(): DiscordModuleOption {
     return {
-      token: process.env.DISCORD_CLIENT_TOKEN,
+      token: process.env.DISCORD_CLIENT_TOKEN ?? '',
```
Comment (Meierschlumpf) [inline src/clients/discord/discord.config.service.ts:12]: would it be possible to just throw an exception that tells the dev that he should define the client token

File: src/clients/discord/discord.voice.service.ts [hunk 2]
```diff
@@ -104,6 +104,10 @@ export class DiscordVoiceService {
   }
 
   changeVolume(volume: number) {
+    if (!this.audioResource || !this.audioResource.volume) {
+      this.logger.error(`AudioResource or volume was undefined`);
```
Comment (Meierschlumpf) [inline src/clients/discord/discord.voice.service.ts:108]: Do you not want to tell the user, that something went wrong?

File: src/clients/jellyfin/jellyfin.service.ts [hunk 3]
```diff
@@ -33,18 +33,23 @@ export class JellyfinService {
       },
     });
 
-    this.api = this.jellyfin.createApi(process.env.JELLYFIN_SERVER_ADDRESS);
+    this.api = this.jellyfin.createApi(
+      process.env.JELLYFIN_SERVER_ADDRESS ?? '',
```
Comment (Meierschlumpf) [inline src/clients/jellyfin/jellyfin.service.ts:37]: throw an error as well

File: src/clients/jellyfin/jellyfin.service.ts [hunk 4]
```diff
@@ -33,18 +33,23 @@ export class JellyfinService {
...
       .authenticateUserByName(
-        process.env.JELLYFIN_AUTHENTICATION_USERNAME,
+        process.env.JELLYFIN_AUTHENTICATION_USERNAME ?? '',
         process.env.JELLYFIN_AUTHENTICATION_PASSWORD,
       )
       .then(async (response) => {
-        if (response.data.SessionInfo === undefined) {
+        if (
+          response.data.SessionInfo === undefined ||
+          response.data.SessionInfo.UserId === undefined
```
Comment (Meierschlumpf) [inline src/clients/jellyfin/jellyfin.service.ts:51]: change to `response.data.SessionInfo?.UserId === undefined`

File: src/commands/play/play.comands.ts [hunk 5]
```diff
@@ -120,7 +120,7 @@ export class PlayItemCommand {
             if (!remoteImage) {
               return embedBuilder;
             }
-            return embedBuilder.setThumbnail(remoteImage.Url);
+            return embedBuilder.setThumbnail(remoteImage.Url ?? '');
```
Comment (Meierschlumpf) [inline src/commands/play/play.comands.ts:123]: Maybe use a default image

File: src/commands/play/play.comands.ts [hunk 6]
```diff
@@ -105,8 +105,8 @@ export class PlayItemCommand {
     this.playbackService.getPlaylistOrDefault().enqueueTracks(tracks);
 
     const remoteImage: RemoteImageInfo | undefined = tracks
-      .flatMap((x) => x.getRemoteImages())
-      .find((x) => true);
+      .flatMap((track) => track.getRemoteImages())
+      .find(() => true);
```
Comment (Meierschlumpf) [inline src/commands/play/play.comands.ts:109]: this looks stupid 😂

**Review by Meierschlumpf — CHANGES_REQUESTED** [review]

**Pull request merged**